src/zenml/zen_stores/schemas/step_run_schemas.py

```diff
@@ -301,7 +301,7 @@ class StepRunInputArtifactSchema(SQLModel, table=True):
     artifact: ArtifactSchema = Relationship(
         back_populates="input_to_step_runs"
     )
-    name: str
+    name: str = Field(nullable=False, primary_key=True)
```

Comment:
Haven't fully grasped the meaning of this yet, but just wanted to point out that there is also the `StepRunOutputArtifactSchema` that might require a similar change.

Reply:
Had more time to think about this and find an answer by myself. There is no way to use the same name for multiple output artifacts generated by the same step run, is there ? First, because you can't define a function to return e.g. `Output[first = dict, second = str, first = dict]` and second, the whole idea makes no practical sense (why would you even want to return the same artifact multiple times ?).

So forget I asked :)

Reply:
You got it exactly right 👍 

Maybe some more explanation what my change does for the input table: Previously we used just the `step_run_id` and `artifact_id` as primary keys for the table that relates steps with their input artifacts. If a step now had two inputs (with different names) that refer to the same artifact, only one of those would be stored in the database, and when then trying to resolve the inputs for the step we would get the `KeyError` that no artifact for input X exists. This PR simply includes the `name` column in the primary key.